core/api/src/modules/ops/app.ts

```diff
@@ -103,4 +132,9 @@ export namespace AppOps {
 
     return { success, message, error };
   }
+
+  async function getConnectionLock(app: App) {
+    const key = `app:${app.guid}:connection:${id}:`;
```

Comment:
wonder if we should make a `objectLock` like we did for `objectCache`

Reply:
Possibly! This one's a little special though, as it's a lock key per node in the cluster.  `id` is something like `hostname+pid`